esphome/components/rc522/__init__.py

```diff
@@ -39,3 +40,6 @@ async def setup_rc522(var, config):
         trigger = cg.new_Pvariable(conf[CONF_TRIGGER_ID])
         cg.add(var.register_trigger(trigger))
         await automation.build_automation(trigger, [(cg.std_string, "x")], conf)
+
+    if CONF_GAIN in config:
+        cg.add(var.set_gain(config[CONF_GAIN]))
```

Comment:
Default in schema means the key is always there
```suggestion
    cg.add(var.set_gain(config[CONF_GAIN]))
```